pytorch_lightning/loggers/wandb.py

```diff
@@ -142,6 +149,10 @@ def __getstate__(self):
         state['_experiment'] = None
         return state
 
+    def connect(self, trainer: 'Trainer') -> None:
+        if trainer is not None:
+            self._trainer = proxy(trainer)
```

Comment:
Should we maybe move the creation of the proxy already to the connector? That way we can ensure that no logger can actually bind to the real trainer object but only to the proxy.